File: PlusMinus/plus-minus.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the plusMinus function below.
function plusMinus(arr, n) {
    console.log(((arr.filter(i => i > 0)).length/ n).toPrecision(6));
    console.log(((arr.filter(i => i < 0)).length/ n).toPrecision(6));
    console.log(((arr.filter(i => i == 0)).length/ n).toPrecision(6));
}

function main() {
    const n = parseInt(readLine(), 10);

    const arr = readLine().split(' ').map(arrTemp => parseInt(arrTemp, 10));

    plusMinus(arr, n);
}

File: PlusMinus/plus-minus.py
import math
import os
import random
import re
import sys

# Complete the plusMinus function below.
def plusMinus(arr,n):
    print ("{0:.6f}".format(len([i for i in arr if i > 0]) / n))
    print ("{0:.6f}".format(len([i for i in arr if i < 0]) / n))
    print ("{0:.6f}".format(len([i for i in arr if i == 0]) / n))
    
if __name__ == '__main__':
    n = int(input())

    arr = list(map(int, input().rstrip().split()))

    plusMinus(arr,n)
